### ISSUE TITLE
[Persist] `time:Utc` type values mismatch on retrieval

### ISSUE THREAD

**Issue by kaneeldias** [ballerina-platform/ballerina-standard-library#4081]
**Description:**
When a `time:Utc` type is persisted and retrieved, the value obtained is different.

**Steps to reproduce:**
```ballerina
type TestEntity record {|
    readonly int id;
    time:Utc utcType;
|};
```

```ballerina
public function main() returns error? {
    time:Utc insertUtc = check time:utcFromString("2007-12-03T10:15:30.00Z");
    test_entities:TestEntity insert = {
        id: 5,
        utcType: insertUtc
    };

    test_entities:TestEntitiesClient teClient = check new();
    _ = check teClient->/testentity.post([insert]);

    test_entities:TestEntity retrieved = check teClient->/testentity/[5].get();
    test:assertEquals(insertUtc, retrieved);
}
```

Output
```
error: Assertion Failed!
 
expected: <TestEntity> '{"id":5,"utcType":[1196657130,0]}'
actual  : <[int,decimal] & readonly> '[1196676930,0]'
```

The difference is 19800, which corresponds to 5.5 hours, indicating this may be an issue with time zones

**Comment by kalaiyarasiganeshalingam** [comment]
The same value is stored in the database, but the problem is in converting SQL to the ballerina type when executing the read command.  In the SQL, the timestamp[1] is converted to milliseconds by using `getTime()`[2]. 

The `getTime()` value of the above sample is `1196657130` and the equivalent time is `2007-12-03T04:45:30Z`. This time is equals to  `Given time - 5.30`.

[1] <img width="588" alt="Screenshot 2023-02-28 at 21 48 08" src="https://user-images.githubusercontent.com/12771261/221914780-8f9e624e-d890-4998-af0a-48dbb514b3ca.png">
[2] https://github.com/ballerina-platform/module-ballerina-sql/blob/master/native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java#L550

@niveathika @daneshk Is it expected behavior?

**Comment by niveathika** [comment]
This is the correct behaviour from the `sql` module perspective. AS `sql` module only stores and retrieves data and does not manipulate this. 

IMO this is not suitable for `persist` module as this is the data layer of an application. Shall we check how this is handled in other ORM libraries?

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam According to your comment,

There is no issue with the value when storing it in the database. the database stores the same value we passed to the persist module and also the internal SQL module.
The problem is when retrieving the data from the database. database returns the correct GMT value without local offset, but the value returned from the SQL connector has local offset. it converts GMT time to the local time when we are converting the time value to milliseconds.

is this correct?

**Comment by kalaiyarasiganeshalingam** [comment]
Yes, there is no issue with the value when storing it in the database and retrieving the data from the database.

The problem is in converting the retrieved data.
The given time `2007-12-03T10:15:30.00Z` in Millis = `1196676930`. 
But, in the SQL, `timestamp.getTime()` returns time in Millis = `119665713`(GMT -5.30). 

If it returns GMT time to local time, it should return the value by adding 5.30 hours (GMT +5.30) but it subtracts that value(GMT -5.30).
Please correct me if I am wrong, it considers this value a local time and converts it to GMT time. So, it subtracts that value by 5.30. But users will want to get back the same value they saved. But neither the GMT time nor the corresponding local time is returned for the given time.

**Comment by daneshk** [comment]
We need to correct this at the SQL level to retrieve the same value.
